File: validation/src/main/scala/hl7/v2/validation/vs/ComplexElementValidator.scala
package hl7.v2.validation.vs

import hl7.v2.instance.Complex
import hl7.v2.validation.report.VSEntry

trait ComplexElementValidator {

  /**
    * Checks the complex element against the value specifications
    * and returns the list of problems detected.
    */
  //FIXME: Update the instance model to differentiate between a data element ...
  // and complex element.
  def check(e: Complex, library: Map[String, ValueSet]): List[VSEntry]
}

File: validation/src/main/java/hl7/v2/validation/vss/impl/ComplexElementValidator.java
package hl7.v2.validation.vss.impl;

import gov.nist.validation.report.Entry;
import hl7.v2.instance.Complex;
import hl7.v2.profile.ValueSetSpec;
import hl7.v2.validation.vss.ValueSetLibrary;

/**
 * Module for validating complex element against a value set specification
 *
 * @author Salifou Sidi M. Malick <salifou.sidi@gmail.com>
 */
public class ComplexElementValidator {

    /**
     * Checks the simple element against the value set specification
     * and a detection if a problem if found, null otherwise
     * @param e       - The complex element to be validated
     * @param spec    - The value set specification
     * @param library - The value set library to be used
     * @return A detection if a problem is found, null otherwise
     */
    public static Entry check(Complex e, ValueSetSpec spec, ValueSetLibrary library) {
        if( spec == null )
            return null;

        return null; //FIXME;
    }
}
